docs/repositories.md

```diff
@@ -153,6 +153,33 @@ processes dealing with the same aggregate would run into concurrency issues very
 The test case has some more tests including snapshot usage and working with different stream names / strategies.
 Just browse through the test methods for details.
 
+## Aggregate Type Mapping
+
+It's possible to map an aggregate type `user` to an aggregate root class like `My\Model\User`. To do that, add the
+aggregate type mapping to your repository and use the provided aggregate type. The aggregate type mapping is the last
+argument of the aggregate repositories constructor.
+
+Example configuration:
+
+```php
+[
+    'prooph' => [
+        'event_sourcing' => [
+            'aggregate_repository' => [
+                'user_repository' => [ //<-- here the container id is referenced
+                    'repository_class' => MyUserRepository::class,
+                    'aggregate_type' => 'user', //<-- The custom aggregate type
```

Comment:
this would be nice:

```php
'aggregate_type' => ['user' => MyUserClass],
```

With `AggregateType::fromArray(['user' => MyUserClass]) and `$aggregateType->mappedClass()`
But I'm not sure if this is a BC break. 

Or maybe `AggregateType::fromString|fromAggregate|fromAggregateRootClass('user', ['user' => MyUserClass, 'todo' => MyTodoClass])`

You have a global aggregateTypeMap and you can pass it (new optional arg) to the AggregateType value object so that mapping is done insight the VO.

Just some thoughts for improvements because repo construct takes already a lot of optional args